tests/system/feature_store/test_feature_store.py

```diff
@@ -3252,6 +3252,328 @@ def test_pandas_stats_include_index(self, index_columns):
 
         assert_frame_equal(expected_stat, actual_stat)
 
+    @pytest.mark.parametrize("with_indexes", [True, False])
+    @pytest.mark.parametrize("engine", ["local", "dask"])
+    @pytest.mark.parametrize("join_type", ["inner", "outer"])
+    def test_relation_join(self, engine, join_type, with_indexes):
+        """Test 3 option of using get offline feature with relations"""
+        departments = pd.DataFrame(
+            {
+                "d_id": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+                "name": [
+                    "dept1",
+                    "dept2",
+                    "dept3",
+                    "dept4",
+                    "dept5",
+                    "dept6",
+                    "dept7",
+                    "dept8",
+                    "dept9",
+                    "dept10",
+                ],
+                "manager_id": [i for i in range(10, 20)],
+            }
+        )
+
+        employees = pd.DataFrame(
+            {
+                "id": [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000],
+                "name": [
+                    "name1",
+                    "name2",
+                    "name3",
+                    "name4",
+                    "name5",
+                    "name6",
+                    "name7",
+                    "name8",
+                    "name9",
+                    "name10",
+                ],
+                "department_id": [1, 1, 1, 2, 2, 2, 6, 6, 6, 11],
+            }
+        )
+
+        employees_mini = pd.DataFrame(
```

Comment:
employees_mini is bad name. please rename employees to employees_with_department, and employees_mini to employees_with_class